### ISSUE TITLE
 Ambiguous Indirect Export error when using elastic-builder in nuxt layer 

### ISSUE THREAD

**Issue by thibaultrey** [nuxt/nuxt#21714]
### Environment

- Operating System: Linux
- Node Version:     v16.14.2                                                                                                                             12:48:46
- Nuxt Version:     3.5.3
- Nitro Version:    2.5.1
- Package Manager:  yarn@1.22.5
- Builder:          vite
- User Config:      extends, vite
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://github.com/thibaultrey/nuxt-extends-import-bug

### Describe the bug

If you import [elastic-builder](https://www.npmjs.com/package/elastic-builder) package in a layer extended as a NPM package, you get the following error when load the page :
`
Uncaught SyntaxError: ambiguous indirect export: default
`

This error doesn't appear when you using the same layer locally via a relative path.

The error seems similar to [#19267](https://github.com/nuxt/nuxt/issues/19267) but added elastic-builder package in the build.transpile settings doesn't works.

In the relative path extend way vite seems to _optimize_ the elastic-builder; this action never happens when you extends throught a NPM Package.

**Comment by thibaultrey** [comment]
Hi,

I just found out that the same error happend with the [nuxt-storage](https://nuxt.com/modules/storage) module imported via `import nuxtStorage from 'nuxt-storage'`

**Error**
`Uncaught SyntaxError: ambiguous indirect export: default `

I found a **temporary** way to fix the error :
1. Installing the problematic package to the Nuxt app that use the layer.
2. Adding the package name to the alias option in the nuxt config file.


I guess something is missing in my process to create a package from an existing app.

**Comment by chrisfoss1** [comment]
I originally opened [19267](https://github.com/nuxt/nuxt/issues/19267).  I just noticed I had this issue occur again with [apollo-upload-client](https://github.com/jaydenseric/apollo-upload-client), and setting transpile didn't work as a solution for that either.

I ended up using `lodash-es` as my import in the original ticket, so I only did a quick check to make sure the transpile solution would work as well before closing it.  It is possible I messed up while checking the solution.  Perhaps the package I pushed up still had the `lodash-es` import?  I will check on that when I have a chance soon.
